### ISSUE TITLE
[HOLD for payment 2024-11-07] [CVP] [$250] If you click "Pay with Expensify" and the payment fails (e.g., due to a bank account setup error), the option disappears, leaving only "Pay Elsewhere."

### ISSUE THREAD

**Issue by m-natarajan** [Expensify/App#49576]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 9.0.39-0
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** @ryanschaffer 
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1726864011581859

## Action Performed:

1. Create new test account.
2. Submit an expense to a new account that hasn’t been created yet
3. Log in to account 2 and try to pay the expense submitted from account 1
4. Mess up the bank flow
5. Try to restart the payment flow and see that the only option is `Pay Elsewhere`
## Expected Result:
If you try to restart the payment flow a 2nd time, it works the same it did the first time.
## Actual Result:
The 2nd time, you don’t get the `Pay with Expensify` option and only the `Pay Elsewhere` option is available.
## Workaround:
Unknown
## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [X] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

![image](https://github.com/user-attachments/assets/4ad43cb1-d261-4d7e-ba8c-ea6277f79d3c)

https://github.com/user-attachments/assets/05e88105-baa0-41ad-995e-04105b40df5d

Add any screenshot/video evidence

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~021838883742527613450</li>
        <li>Upwork Job ID: 1838883742527613450</li>
        <li>Last Price Increase: 2024-09-25</li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @joekaufmanexpensify</details>

**Comment by Ace0328** [comment]
To resolve this issue, you should ensure that the resendTransaction method is correctly defined and implemented in the appropriate file within your codebase, and that the import statement or class extension is set up correctly if it is inherited from a parent class or imported from a library. You should also ensure that there are no typos or errors in the method name, and that all required dependencies are correctly installed and referenced.

**Comment by Ace0328** [comment]
***@***.***
Upwork Profile link:https://www.upwork.com/freelancers/~017675b79a86fc30f6

**Comment by shahinyan11** [comment]
@m-natarajan Hi. The issue does not reproducible with the steps described in **Action Performed:** section. There is not a button with the "Pay $xx with Expensify" label .  Could you provide a video showing the entire flow of actions?

**Comment by mananjadhav** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.

We're disabling the wallet options for the expense report. Hence we don't see "Pay with Expensify" for the workspace payments.

### What is the root cause of that problem?

After clicking the "Pay with Expensify" option and selecting the "Business Bank Account",  we move the payment to a new workspace and the payment type becomes `CONST.REPORT.TYPE.EXPENSE`. The "Pay with Expensify" option is dependent on the "canUseWallet" as shown in the code below. 

https://github.com/Expensify/App/blob/43c2409e5e740cfd52f2b4a43b00957e3dbe25c2/src/components/SettlementButton/index.tsx#L98-L108

The flag `canUseWallet` is false `isExpenseReport` is true which is in our case.


### What changes do you think we should make in order to solve the problem?

- Remove `isExpenseReport` check from the condition. The flag will now be conditioned as:

```
const canUseWallet = !isInvoiceReport && currency === CONST.CURRENCY.USD;
```

### What alternative solutions did you explore? (Optional)

**Comment by ryanschaffer** [comment]
I just retested this. Another way to make the `Pay with Expensify` option disappear is to just exit out of the Plaid flow. I was testing this flow, realized I wasn't on staging so reloaded on staging and `Pay Elsewhere` was the only option

**Comment by allroundexperts** [comment]
@mananjadhav's [proposal](https://github.com/Expensify/App/issues/49576#issuecomment-2375083441) looks good to me. However, before proceeding, I'd like the assigned engineer to confirm that the user can use the wallet if its an expense report?

🎀 👀 🎀 C+ reviewed

**Comment by trjExpensify** [comment]
Hm, I don't think the wallet has anything to do with expense reports, that's a P2P IOUReport feature. 

I actually think the crux of the problem is that the workspace created has `reimbursementChoice=manual` on creation because a VBBA hasn't been added to the workspace successfully yet to set that to `reimbursementChoice=yes` for in-app payments on workspaces.

So we need to decide if we want to show the "Pay with Expensify" button in the dropdown for USD workspaces if `reimbursementChoice=manual`. If that option in the pay button is clicked, launch the connect bank account VBBA flow, and when successfully added it will update to `reimbursementChoice=yes`.

**Comment by JmillsExpensify** [comment]
> Hm, I don't think the wallet has anything to do with expense reports, that's a P2P IOUReport feature.

Agreed

> So we need to decide if we want to show the "Pay with Expensify" button in the dropdown for USD workspaces if reimbursementChoice=manual. If that option in the pay button is clicked, launch the connect bank account VBBA flow, and when successfully added it will update to reimbursementChoice=yes.

Also agree here. Let's do it.

**Comment by neil-marcellini** [comment]
Ok. @allroundexperts @mananjadhav would you please investigate Tom's theory for the root cause and come up with a solution that doesn't relate to changing the canUseWallet permissions? Maybe that's actually a poor variable name in the code? Please update your proposal and let us know.

We are still seeking new proposals.

**Comment by mananjadhav** [comment]
I don't have the context for this, but I'll look up and propose a solution.

**Comment by dylanexpensify** [comment]
TY @mananjadhav!

**Comment by Ace0328** [comment]
Haha, thanks!

**Comment by dylanexpensify** [comment]
@mananjadhav any updates?

**Comment by mananjadhav** [comment]
Okay I tried following @trjExpensify's comment [here](https://github.com/Expensify/App/issues/49576#issuecomment-2387123829). I was unnecessarily complicating it thinking we need to update the reimbursementChoice when we start the bank flow.

> So we need to decide if we want to show the "Pay with Expensify" button in the dropdown for USD workspaces if reimbursementChoice=manual. 

If we're good with this, then the only change we need to make is this [line](https://github.com/Expensify/App/blob/5f5688f02cd53fec9d3749f341203b52f8de1a1c/src/components/SettlementButton/index.tsx#L72)

We should remove `policy?.reimbursementChoice !== CONST.POLICY.REIMBURSEMENT_CHOICES.REIMBURSEMENT_MANUAL` check and that should work. I'll write up a formal proposal.

**Comment by mananjadhav** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.

We're disabling the wallet options in the workspace chat. Hence we don't see "Pay with Expensify".

### What is the root cause of that problem?

After clicking the "Pay with Expensify" option and selecting the "Business Bank Account", we move the payment to a new workspace. By default the `reimbursementChoice` is manual for the workspace. In the code below, Pay with Expensify option is hidden if the `reimbursementChoice` is manual..

https://github.com/Expensify/App/blob/5f5688f02cd53fec9d3749f341203b52f8de1a1c/src/components/SettlementButton/index.tsx#L72

### What changes do you think we should make in order to solve the problem?

> So we need to decide if we want to show the "Pay with Expensify" button in the dropdown for USD workspaces if reimbursementChoice=manual.

Based on the previous comment here, then the only change we need to make is this [line](https://github.com/Expensify/App/blob/5f5688f02cd53fec9d3749f341203b52f8de1a1c/src/components/SettlementButton/index.tsx#L72)

We should remove `policy?.reimbursementChoice !== CONST.POLICY.REIMBURSEMENT_CHOICES.REIMBURSEMENT_MANUAL` check and that should work.  I have attached the video in the previous comment.



### What alternative solutions did you explore? (Optional)

**Reminder:** Please use plain English, be brief and avoid jargon. Feel free to use images, charts or pseudo-code if necessary. Do not post large multi-line diffs or write walls of text. Do not create PRs unless you have been hired for this job.

**Comment by trjExpensify** [comment]
When `REIMBURSEMENT_NO` is the `reimbursementChoice` for say a track user just categorising their receipts on a workspace, wouldn't we end up showing the `Pay with Expensify` button or is that logic elsewhere?

**Comment by mananjadhav** [comment]
tl;dr, `REIMBURSEMENT_NO` is taken care of by evaluating outside the component and passed in as props.

**Explanation:**

The settlement button payment options has two conditions. One about manual that I mentioned, second is the prop `shouldHidePaymentOptions`

https://github.com/Expensify/App/blob/7746ea31493d47aa779797e1df1e0e76d2d6ed0d/src/components/SettlementButton/index.tsx#L72

If we check the props being passed in `ReportPreview`, you'll see it's dependent on `shouldShowPayButton`.

https://github.com/Expensify/App/blob/7746ea31493d47aa779797e1df1e0e76d2d6ed0d/src/components/ReportActionItem/ReportPreview.tsx#L534

The `shouldShowPayButton` is then [defined](https://github.com/Expensify/App/blob/7746ea31493d47aa779797e1df1e0e76d2d6ed0d/src/components/ReportActionItem/ReportPreview.tsx#L324) using `canIOUBePaid` where the check for `REIMBURSEMENT_NO` exists.

https://github.com/Expensify/App/blob/7746ea31493d47aa779797e1df1e0e76d2d6ed0d/src/libs/actions/IOU.ts#L6992-L6999

**Comment by Ace0328** [comment]
Thanks, I'll check it out!

**Comment by trjExpensify** [comment]
>tl;dr, REIMBURSEMENT_NO is taken care of by evaluating outside the component and passed in as props.

Great then!

**Comment by trjExpensify** [comment]
@allroundexperts can you progress the review today? Let's get this fixed!

**Comment by allroundexperts** [comment]
Sure thing!

**Comment by allroundexperts** [comment]
My day hasn't finished yet 😄. I'll review this shortly in about 2 hours.

**Comment by allroundexperts** [comment]
Okay. I managed to check @mananjadhav's [proposal](https://github.com/Expensify/App/issues/49576#issuecomment-2411985804). Given we have an agreement that we should show the `Pay with Expensify` regardless if the reimbursementChoice is manual or not, @mananjadhav's proposal of removing the condition should work well. Let's go with there proposal.

🎀 👀 🎀 C+ reviewed

**Comment by dylanexpensify** [comment]
Hey @joekaufmanexpensify! I'm heading out on parental leave so reassigning this! TY!

**Comment by joekaufmanexpensify** [comment]
Sure, happy to help!

**Comment by joekaufmanexpensify** [comment]
I think we're just waiting for @neil-marcellini to sign off on the selected proposal above

**Comment by joekaufmanexpensify** [comment]
Dm'ed Neil about proposal

**Comment by neil-marcellini** [comment]
> Okay. I managed to check @mananjadhav's [proposal](https://github.com/Expensify/App/issues/49576#issuecomment-2411985804). Given we have an agreement that we should show the `Pay with Expensify` regardless if the reimbursementChoice is manual or not, @mananjadhav's proposal of removing the condition should work well. Let's go with there proposal.
> 
> 🎀 👀 🎀 C+ reviewed

Sounds good to me.

**Comment by mananjadhav** [comment]
Thanks, will raise the PR.

**Comment by joekaufmanexpensify** [comment]
Great. TY!

**Comment by joekaufmanexpensify** [comment]
Pending PR

**Comment by mananjadhav** [comment]
Will be pushed in 2-3 hours, just testing changes locally.

**Comment by joekaufmanexpensify** [comment]
Sounds great. TY!

**Comment by joekaufmanexpensify** [comment]
PR in review

**Comment by joekaufmanexpensify** [comment]
@allroundexperts could you tackle the checklist this week so I can prep for payment?

**Comment by joekaufmanexpensify** [comment]
Great. TY! I will action.

**Comment by joekaufmanexpensify** [comment]
Regression test created: https://github.com/Expensify/Expensify/issues/442826

**Comment by joekaufmanexpensify** [comment]
All set to issue payment. We need to pay:

- @mananjadhav - $250 for PR via NewDot. 
- @allroundexperts - $250 for C+ via NewDot. 

Feel free to request payment as soon as you're both ready!

**Comment by joekaufmanexpensify** [comment]
Closing for now, as both contributors will request payment whenever they are ready to. Thanks everyone!

**Comment by JmillsExpensify** [comment]
$250 approved for @allroundexperts
